bootloader/main.c

```diff
@@ -408,20 +437,39 @@ get_real_prog_path(void)
     return result;
 }
 
+char *get_tmp_root(void)
+{
+    char *root_dir;
+    root_dir = TMPROOTDIR;
+
+    char *ret;
+	ret = (char*) malloc (strlen(root_dir)+1);
+	strcpy(ret, root_dir);
+    ret = strtrim(ret);
+    return ret;
+}
 static void identify(void)
 {
     debug_printf("bootloader version %s\n", STATICX_VERSION);
     debug_printf("compiled %s at %s by %s version %s\n",
             __DATE__, __TIME__, COMPILER_PATH, __VERSION__);
 
-    /* If we're invoked by staticx, just exit */
     if (getenv("STATICX_BOOTLOADER_IDENTIFY"))
         exit(0);
 }
+static void set_tmpdir(void)
+{
+    const char *tmproot = get_tmp_root();
+    debug_printf("tmproot: %s\n", tmproot);
 
+    if (strstr(tmproot, "tmp_folder_define") == NULL) {
+        setenv(TMPDIR, tmproot, 0);
```

Comment:
As implemented:
* `--tmprootdir` ignores `TMPDIR` set in the target environment.
* `--tmprootdir` also affects `TMPDIR` for the child process (not just the staticx bootloader)

Is this intentional?